document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/parser/filter/PostgresRelationalFilterParser.java

```diff
@@ -0,0 +1,28 @@
+package org.hypertrace.core.documentstore.postgres.query.v1.parser.filter;
+
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.Value;
+import lombok.experimental.Accessors;
+import lombok.experimental.Delegate;
+import org.hypertrace.core.documentstore.expression.impl.RelationalExpression;
+import org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser;
+import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresConstantExpressionVisitor;
+import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresSelectTypeExpressionVisitor;
+
+public interface PostgresRelationalFilterParser {
+  String parse(
```

Comment:
Will every parse request have a different context? If not, shall we consider moving it as part of the Parser constructor?

Reply:
> Will every parse request have a different context?

Yes. That's the idea in the continuation PR @ https://github.com/hypertrace/document-store/pull/191